core/src/main/java/org/testcontainers/dockerclient/WslNpipeSocketClientProviderStrategy.java

```diff
@@ -0,0 +1,77 @@
+package org.testcontainers.dockerclient;
+
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientConfig;
+import java.net.URI;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Delegate;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.SystemUtils;
+import org.jetbrains.annotations.NotNull;
+
+@Slf4j
+public class WslNpipeSocketClientProviderStrategy extends DockerClientProviderStrategy {
```

Comment:
This class is a duplicate of `NpipeSocketClientProviderStrategy`, would be much better to extend it (either from an abstract common class or from the same)

Reply:
Definitely 👍
I stopped because I ran into the error and decided to draft what I currently had 🤷‍♂